pkg/apis/operator/v1/authentication_types.go

```diff
@@ -0,0 +1,92 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type AuthMethod string
+
+const (
+	AuthMethodOIDC AuthMethod = "OIDC"
+)
+
+// AuthenticationSpec defines the desired state of Authentication
+type AuthenticationSpec struct {
+	// Method configures the method of authentication used by Kibana.
+	// Default: Basic
+	// +kubebuilder:validation:Enum=Basic;OIDC
+	// +required
+	Method AuthMethod `json:"method,omitempty"`
+
+	// ManagerDomain is the domain name of the Manager
+	// +required
+	ManagerDomain string `json:"managerDomain,omitempty"`
+
+	// OIDC contains the configuration needed to setup OIDC authentication. If the method is OIDC then this is required, if
+	// the method is not OIDC then this must not be specified.
+	// +optional
+	OIDC *AuthenticationOIDC `json:"oidc"`
+}
+
+// AuthenticationStatus defines the observed state of Authentication
+type AuthenticationStatus struct {
+	// State provides user-readable status.
+	State string `json:"state,omitempty"`
+}
+
+// AuthenticationOIDC is the configuration needed to setup OIDC.
+type AuthenticationOIDC struct {
+	// IssuerURL is the URL to the OIDC provider.
+	// +required
+	IssuerURL string `json:"issuerURL"`
+
+	// UsernameClaim specifies which claim to use from the OIDC provider as the username.
+	// +required
+	UsernameClaim string `json:"usernameClaim"`
+
+	// RequestedScopes is a list of scopes to request from the OIDC provider. If not provided, all the available scopes
+	// are requested.
+	RequestedScopes []string `json:"requestedScopes"`
```

Comment:
If not provided you get a validation error. Maybe you meant to make this optional as well?